front/app/resources/GetIdeas.tsx

```diff
@@ -79,427 +48,185 @@ export type GetIdeasChildProps = State & {
-      'page[number]': props.pageNumber as number,
-      'page[size]': props.pageSize as number,
-      phase: props.phaseId,
-      author: props.authorId,
-      sort: props.sort as Sort,
-      search: props.search,
-      topics: props.topics,
-      idea_status: props.ideaStatusId,
-      publication_status: props.publicationStatus,
-      project_publication_status: props.projectPublicationStatus,
-      bounding_box: props.boundingBox,
-      assignee: props.assignee,
-      feedback_needed: props.feedbackNeeded,
-      filter_can_moderate: props.filterCanModerate,
-      basket_id: props.basketId,
-    };
-
-    // Omit all queryParameters that are nil.
-    // Why do this? Because we assume that an input prop that's nil is an input prop that should be ignored,
-    // and not overwrite a none-nil value that's part of this.state.queryParameters.
-    return {
-      ...queryParameters,
-      ...omitBy(inputPropsQueryParameters, isNil),
-      // Make an exception for 'projects', because when it's undefined we don't want to ignore it but instead pass it along
-      // to let the request know we don't want to apply a projects filter but load the ideas for all projects
-      projects,
-    };
-  };
-
-  loadMore = () => {
-    if (!this.state.loadingMore) {
-      this.queryParameters$.next({
-        ...this.state.queryParameters,
-        'page[number]': this.state.queryParameters['page[number]'] + 1,
-      });
-    }
-  };
-
-  handleChangePage = (pageNumber: number) => {
-    this.queryParameters$.next({
-      ...this.state.queryParameters,
-      'page[number]': pageNumber,
-    });
-  };
-
-  handlePhaseOnChange = (phaseId: string) => {
-    this.queryParameters$.next({
-      ...this.state.queryParameters,
-      phase: phaseId,
-      'page[number]': 1,
-    });
-  };
-
-  handleSearchOnChange = (search: string) => {
-    this.queryParameters$.next({
-      ...this.state.queryParameters,
-      search: search ? search : undefined,
-      'page[number]': 1,
-    });
-  };
-
-  handleSortOnChange = (sort: Sort) => {
-    this.queryParameters$.next({
-      ...this.state.queryParameters,
+  const { data, isFetching, isRefetching } = useIdeas(queryParameters);
+
+  const list = data?.data;
+
+  const currentPage =
+    (data?.links.self ? getPageNumberFromUrl(data?.links?.self) : null) ?? 1;
+
+  const lastPage =
+    (data?.links.last ? getPageNumberFromUrl(data?.links?.last) : null) ?? 1;
+
+  const sortAttribute = getSortAttribute<Sort, SortAttribute>(
+    queryParameters.sort
+  );
+  const sortDirection = getSortDirection<Sort>(queryParameters.sort);
+
+  const updateQuery = useCallback(
+    (newQueryParameters: Partial<IQueryParameters>) => {
+      if (noChanges(newQueryParameters, queryParameters)) return;
+
+      setQueryParameters((currentQueryParameters) => ({
+        ...currentQueryParameters,
+        ...newQueryParameters,
+      }));
+    },
+    [queryParameters]
+  );
+
+  const loadMore = useCallback(() => {
```

Comment:
I thought we deleted the infinite query variant from here. This means this is no longer needed, correct?